### PR TITLE
Let user download .ics file if they don't have Outlook Calendar (MVP Stretch)

### PR DESCRIPTION
**Pull request by Calebc4554**
# Description  
- What does this PR do?  
  Adds support for users to download an `.ics` calendar file for any event, enabling calendar integration for users who do not use Outlook or Google Calendar.  
- Why is it necessary or valuable?  
  Some users may not use supported calendar integrations. This feature ensures broader accessibility by allowing any user to add events to their calendar using the industry-standard `.ics` format.  

---

# Milestones / Related Work  
- Milestone or version target: 
[  - 5.24 ICS file Implementation (Stretch MVP)](https://docs.google.com/document/d/1gKQkyru2JwfG13o2M-rWdAOgh8Qr6j2nxwepipoh564/edit?tab=t.0#bookmark=kix.7djx06dtmfw7)

---

# Resources and References  
- Tutorials, documentation, or code examples used:  
  - https://icalendar.org/iCalendar-RFC-5545/3-6-1-event-component.html  

---

# Test Plan  
- How was this tested?  
  Manually verified `.ics` downloads across multiple events, checked file format opens in native calendar apps (Apple Calendar, Outlook, etc.).  
  Added unit test in `Events.test.jsx` to check for presence of download button.  
- Screenshots:  
Hackathon Build for Social Good

## Edge Cases Tested  
- Events with missing location  
- Events with no description  
- Events with special characters in title (cleaned for filename)

### CODE REVIEW

**Review by alegleason — COMMENTED** [review]
Can you add a demo of the `.ics` file being dowloaded and added to your calendar?

File: frontend/src/__tests__/Events.test.jsx [hunk 1]
```diff
@@ -381,4 +381,22 @@ describe("Events Component", () => {
 			});
 		});
 	});
+
+	describe("Calendar Integration", () => {
+		it("shows download button for ICS files", async () => {
+			renderEvents();
+			await waitFor(() => {
+				const downloadButtons = screen.getAllByText("Download");
+				expect(downloadButtons.length).toBeGreaterThan(0);
```
Comment (alegleason) [inline frontend/src/__tests__/Events.test.jsx:390]: Maybe we can compare the string contents too? :D

File: frontend/src/components/EventDetailModal.jsx [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
 
+const generateICSFile = (event) => {
+	const formatDate = (date) => {
+		return new Date(date).toISOString().replace(/[-:]/g, "").split(".")[0] + "Z";
+	};
+
+	const eventDate = new Date(event.date);
+	const endDate = new Date(eventDate.getTime() + 2 * 60 * 60 * 1000);
+
+	const icsContent = ["BEGIN:VCALENDAR", "VERSION:2.0", "PRODID:-//MICS//Event Calendar//EN", "BEGIN:VEVENT", `UID:${event.id}@mics.colorstack.org`, `DTSTART:${formatDate(eventDate)}`, `DTEND:${formatDate(endDate)}`, `SUMMARY:${event.title}`, `DESCRIPTION:${event.description || "No description available"}`, `LOCATION:${event.location || "No location specified"}`, `ORGANIZER:${event.users?.name || "Unknown"}`, "END:VEVENT", "END:VCALENDAR"].join("\r\n");
```
Comment (alegleason) [inline frontend/src/components/EventDetailModal.jsx:15]: Might this be easier to read?

```
const icsContent = [
  "BEGIN:VCALENDAR",
  "VERSION:2.0",
  "PRODID:-//MICS//Event Calendar//EN",
  "BEGIN:VEVENT",
  `UID:${event.id}@mics.colorstack.org`,
  `DTSTART:${formatDate(eventDate)}`,
  `DTEND:${formatDate(endDate)}`,
  `SUMMARY:${event.title}`,
  `DESCRIPTION:${event.description || "No description available"}`,
  `LOCATION:${event.location || "No location specified"}`,
  `ORGANIZER:${event.users?.name || "Unknown"}`,
  "END:VEVENT",
  "END:VCALENDAR"
].join("\r\n");
```

File: frontend/src/components/Events.jsx [hunk 3]
```diff
@@ -33,6 +33,32 @@ function Modal({ open, title, message, onClose, onConfirm, confirmText = "OK", c
...
+		return new Date(date).toISOString().replace(/[-:]/g, "").split(".")[0] + "Z";
+	};
+
+	const eventDate = new Date(event.date);
+	const endDate = new Date(eventDate.getTime() + 2 * 60 * 60 * 1000); // Default 2 hours duration
+
+	const icsContent = ["BEGIN:VCALENDAR", "VERSION:2.0", "PRODID:-//MICS//Event Calendar//EN", "BEGIN:VEVENT", `UID:${event.id}@mics.colorstack.org`, `DTSTART:${formatDate(eventDate)}`, `DTEND:${formatDate(endDate)}`, `SUMMARY:${event.title}`, `DESCRIPTION:${event.description || "No description available"}`, `LOCATION:${event.location || "No location specified"}`, `ORGANIZER:${event.users?.name || "Unknown"}`, "END:VEVENT", "END:VCALENDAR"].join("\r\n");
+
+	return icsContent;
+};
```
Comment (alegleason) [inline frontend/src/components/Events.jsx:47]: Seems like this function is exactly the same than the one defined on `EventDetailModal`. Maybe we can exclude them on a utility  file?
